babel.config.js

```diff
@@ -17,6 +17,7 @@ const defaultAlias = {
   '@mui/x-charts': resolveAliasPath('./packages/x-charts/src'),
   '@mui/x-tree-view': resolveAliasPath('./packages/x-tree-view/src'),
   '@mui/markdown': '@mui/monorepo/packages/markdown',
+  '@mui-internal/api-docs-builder': '@mui/monorepo/packages/api-docs-builder',
```

Comment:
What do you think about the following diff.
It would fix these aliases, allow using them, and replace all relevant instances. 🤔 

```diff
diff --git a/babel.config.js b/babel.config.js
index 4e02caa2e..325952615 100644
--- a/babel.config.js
+++ b/babel.config.js
@@ -16,10 +16,13 @@ const defaultAlias = {
   '@mui/x-date-pickers-pro': resolveAliasPath('./packages/x-date-pickers-pro/src'),
   '@mui/x-charts': resolveAliasPath('./packages/x-charts/src'),
   '@mui/x-tree-view': resolveAliasPath('./packages/x-tree-view/src'),
-  '@mui/markdown': '@mui/monorepo/packages/markdown',
-  '@mui-internal/api-docs-builder': '@mui/monorepo/packages/api-docs-builder',
-  '@mui-internal/docs-utilities': '@mui/monorepo/packages/docs-utilities',
-  'typescript-to-proptypes': '@mui/monorepo/packages/typescript-to-proptypes/src',
+  '@mui/markdown': resolveAliasPath('./node_modules/@mui/monorepo/packages/markdown'),
+  '@mui-internal/api-docs-builder': resolveAliasPath(
+    './node_modules/@mui/monorepo/packages/api-docs-builder',
+  ),
+  '@mui-internal/docs-utilities': resolveAliasPath(
+    './node_modules/@mui/monorepo/packages/docs-utilities',
+  ),
   docs: resolveAliasPath('./node_modules/@mui/monorepo/docs'),
   test: resolveAliasPath('./test'),
   packages: resolveAliasPath('./packages'),
diff --git a/docs/scripts/api/buildComponentsDocumentation.ts b/docs/scripts/api/buildComponentsDocumentation.ts
index 1456ddb6a..9bf2eabef 100644
--- a/docs/scripts/api/buildComponentsDocumentation.ts
+++ b/docs/scripts/api/buildComponentsDocumentation.ts
@@ -6,30 +6,26 @@ import {
 import * as fse from 'fs-extra';
 import fs from 'fs';
 import path from 'path';
-import parseStyles from '@mui/monorepo/packages/api-docs-builder/utils/parseStyles';
+import parseStyles from '@mui-internal/api-docs-builder/utils/parseStyles';
 import fromPairs from 'lodash/fromPairs';
 import createDescribeableProp, {
   DescribeablePropDescriptor,
-} from '@mui/monorepo/packages/api-docs-builder/utils/createDescribeableProp';
-import generatePropDescription from '@mui/monorepo/packages/api-docs-builder/utils/generatePropDescription';
+} from '@mui-internal/api-docs-builder/utils/createDescribeableProp';
+import generatePropDescription from '@mui-internal/api-docs-builder/utils/generatePropDescription';
 import { parse as parseDoctrine } from 'doctrine';
 import generatePropTypeDescription, {
   getChained,
-} from '@mui/monorepo/packages/api-docs-builder/utils/generatePropTypeDescription';
-import parseTest from '@mui/monorepo/packages/api-docs-builder/utils/parseTest';
+} from '@mui-internal/api-docs-builder/utils/generatePropTypeDescription';
+import parseTest from '@mui-internal/api-docs-builder/utils/parseTest';
 import kebabCase from 'lodash/kebabCase';
 import camelCase from 'lodash/camelCase';
 import { LANGUAGES } from 'docs/config';
-import findPagesMarkdownNew from '@mui/monorepo/packages/api-docs-builder/utils/findPagesMarkdown';
+import findPagesMarkdownNew from '@mui-internal/api-docs-builder/utils/findPagesMarkdown';
 import { defaultHandlers, parse as docgenParse } from 'react-docgen';
-import {
-  renderInline as renderMarkdownInline,
-  getHeaders,
-  getTitle,
-} from '@mui/monorepo/packages/markdown';
-import { getLineFeed } from '@mui/monorepo/packages/docs-utilities';
+import { renderInline as renderMarkdownInline, getHeaders, getTitle } from '@mui/markdown';
+import { getLineFeed } from '@mui-internal/docs-utilities';
 import { unstable_generateUtilityClass as generateUtilityClass } from '@mui/utils';
-import type { ReactApi as CoreReactApi } from '@mui/monorepo/packages/api-docs-builder/ApiBuilders/ComponentApiBuilder';
+import type { ReactApi as CoreReactApi } from '@mui-internal/api-docs-builder/ApiBuilders/ComponentApiBuilder';
 import {
   DocumentedInterfaces,
   getJsdocDefaultValue,
diff --git a/docs/scripts/api/buildGridEventsDocumentation.ts b/docs/scripts/api/buildGridEventsDocumentation.ts
index 2c8570d2e..5d4dfdfd4 100644
--- a/docs/scripts/api/buildGridEventsDocumentation.ts
+++ b/docs/scripts/api/buildGridEventsDocumentation.ts
@@ -1,6 +1,6 @@
 import * as ts from 'typescript';
 import path from 'path';
-import { renderInline as renderMarkdownInline } from '@mui/monorepo/packages/markdown';
+import { renderInline as renderMarkdownInline } from '@mui/markdown';
 import {
   DocumentedInterfaces,
   getSymbolDescription,
diff --git a/docs/scripts/api/buildInterfacesDocumentation.ts b/docs/scripts/api/buildInterfacesDocumentation.ts
index 5308c5c43..8634f45c3 100644
--- a/docs/scripts/api/buildInterfacesDocumentation.ts
+++ b/docs/scripts/api/buildInterfacesDocumentation.ts
@@ -2,7 +2,7 @@ import * as ts from 'typescript';
 import * as prettier from 'prettier';
 import kebabCase from 'lodash/kebabCase';
 import path from 'path';
-import { renderInline as renderMarkdownInline } from '@mui/monorepo/packages/markdown';
+import { renderInline as renderMarkdownInline } from '@mui/markdown';
 import {
   escapeCell,
   getSymbolDescription,
diff --git a/docs/scripts/createXTypeScriptProjects.ts b/docs/scripts/createXTypeScriptProjects.ts
index 88de3ba97..c7ed4ceca 100644
--- a/docs/scripts/createXTypeScriptProjects.ts
+++ b/docs/scripts/createXTypeScriptProjects.ts
@@ -3,7 +3,7 @@ import {
   createTypeScriptProject,
   CreateTypeScriptProjectOptions,
   TypeScriptProject,
-} from '@mui/monorepo/packages/api-docs-builder/utils/createTypeScriptProject';
+} from '@mui-internal/api-docs-builder/utils/createTypeScriptProject';
 import { getComponentFilesInFolder } from './utils';
 
 const workspaceRoot = path.resolve(__dirname, '../../');
diff --git a/docs/scripts/generateProptypes.ts b/docs/scripts/generateProptypes.ts
index 4d279b9b7..bd2b66763 100644
--- a/docs/scripts/generateProptypes.ts
+++ b/docs/scripts/generateProptypes.ts
@@ -6,7 +6,7 @@ import {
   getPropTypesFromFile,
   injectPropTypesInFile,
 } from '@mui/monorepo/packages/typescript-to-proptypes';
-import { fixBabelGeneratorIssues, fixLineEndings } from '@mui/monorepo/packages/docs-utilities';
+import { fixBabelGeneratorIssues, fixLineEndings } from '@mui-internal/docs-utilities';
 import { createXTypeScriptProjects, XTypeScriptProject } from './createXTypeScriptProjects';
 
 const prettierConfig = prettier.resolveConfig.sync(process.cwd(), {
diff --git a/tsconfig.json b/tsconfig.json
index 1473b721f..7a670513a 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -26,6 +26,14 @@
       "@mui/x-tree-view/*": ["./packages/x-tree-view/src/*"],
       "@mui/x-license-pro": ["./packages/x-license-pro/src"],
       "@mui/x-license-pro/*": ["./packages/x-license-pro/src/*"],
+      "@mui/markdown": ["./node_modules/@mui/monorepo/packages/markdown"],
+      "@mui/markdown/*": ["./node_modules/@mui/monorepo/packages/markdown/*"],
+      "@mui-internal/api-docs-builder": ["./node_modules/@mui/monorepo/packages/api-docs-builder"],
+      "@mui-internal/api-docs-builder/*": [
+        "./node_modules/@mui/monorepo/packages/api-docs-builder/*"
+      ],
+      "@mui-internal/docs-utilities": ["./node_modules/@mui/monorepo/packages/docs-utilities"],
+      "@mui-internal/docs-utilities/*": ["./node_modules/@mui/monorepo/packages/docs-utilities/*"],
       "test/*": ["./test/*"],
       "docs/*": ["./node_modules/@mui/monorepo/docs"],
       "docsx/*": ["./docs/*"]

```

Reply:
Looks great, I'll do a follow up :ok_hand: